### PR TITLE
Clarify that the "transparency" parameter in plot/plot3d/text can be 1d array 

### LINKED ISSUE #1098: The transparency parameter of plot/plot3d methods can be 1d array
**Description of the problem**

We need to update the documentation for the ` transparency` parameter in [plot](https://www.pygmt.org/dev/api/generated/pygmt.Figure.plot.html#pygmt.Figure.plot) and [plot3d](https://www.pygmt.org/dev/api/generated/pygmt.Figure.plot3d.html#pygmt.Figure.plot3d) because `transparency` can also be `1d array`.

---

See the description  of the `transparency` parameter of [plot](https://www.pygmt.org/dev/api/generated/pygmt.Figure.plot.html#pygmt.Figure.plot):
> **transparency**  (int or float) – Set transparency level, in [0-100] percent range. Default is 0, i.e., opaque. Only visible when PDF or raster format output is selected. Only the PNG format selection adds a transparency layer in the image (for further processing). transparency can also be a 1d array to set varying transparency for symbols.

See the description of the `transparency` parameter of [plot3d](https://www.pygmt.org/dev/api/generated/pygmt.Figure.plot3d.html#pygmt.Figure.plot3d):
> **transparency** (int or float) – Set transparency level, in [0-100] percent range. Default is 0, i.e., opaque. Only visible when PDF or raster format output is selected. Only the PNG format selection adds a transparency layer in the image (for further processing). transparency can also be a 1d array to set varying transparency for symbols.

We can see that `transparency` can be 1d array at the end of the above description, but it is missing at the beginning. See the example of [Points with varying transparency](https://www.pygmt.org/latest/gallery/symbols/points_transparency.html#sphx-glr-gallery-symbols-points-transparency-py) using a 1d array as `transparency`.

---

Meanwhile, array-like `transparency` doesn't work for the `data` parameter. It is only valid for `x`/`y` like the `sizes` parameter. (originally posted in https://github.com/GenericMappingTools/pygmt/pull/1065#discussion_r598100163).

A `GMTInvalidInput` is raised for an array-like `transparency` when the `data` parameter is used in #1065. We may add a test for this kind of invalid input. A test called `test_plot_fail_color_size_intensity` is updated in #1065 for 1d-array `color`, `sizes`, and `intensity` when `data` parameter is used. It should be easy to add `transparency` in this test function, which may have a more general name, e.g.,`test_plot_fail_1d_array_with_data`

### PR DESCRIPTION
**Pull request by core-man**
**Description of proposed changes**

Fixes #1098 

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by core-man** [comment]
I am going to add 1d array transparency test, so the following functions should be renamed to a general one. How about `test_plot_fail_1d_array_with_data` and `test_plot3d_fail_1d_array_with_data`

https://github.com/GenericMappingTools/pygmt/blob/036e2b07170ca7f73c1a034e4da03e17b5250bd4/pygmt/tests/test_plot.py#L94-L98

https://github.com/GenericMappingTools/pygmt/blob/036e2b07170ca7f73c1a034e4da03e17b5250bd4/pygmt/tests/test_plot3d.py#L111-L114

---

I am wondering if we should add **1d array** in the following `COMMON_OPTIONS`. Currently, some modules still do not support 1d array transparency, e.g, [wiggle](https://www.pygmt.org/dev/api/generated/pygmt.Figure.wiggle.html#pygmt.Figure.wiggle), since we may update the alias later.

https://github.com/GenericMappingTools/pygmt/blob/036e2b07170ca7f73c1a034e4da03e17b5250bd4/pygmt/helpers/decorators.py#L128-L129

**Comment by core-man** [comment]
I am wondering if we should add **1d array** in the following `COMMON_OPTIONS`. Currently, some modules still do not support 1d array transparency, e.g, [wiggle](https://www.pygmt.org/dev/api/generated/pygmt.Figure.wiggle.html#pygmt.Figure.wiggle), since we may update the alias later.

https://github.com/GenericMappingTools/pygmt/blob/036e2b07170ca7f73c1a034e4da03e17b5250bd4/pygmt/helpers/decorators.py#L128-L129

**Comment by seisman** [comment]
> I am wondering if we should add **1d array** in the following `COMMON_OPTIONS`. Currently, some modules still do not support 1d array transparency, e.g, [wiggle](https://www.pygmt.org/dev/api/generated/pygmt.Figure.wiggle.html#pygmt.Figure.wiggle), since we may update the alias later.
> 
> https://github.com/GenericMappingTools/pygmt/blob/036e2b07170ca7f73c1a034e4da03e17b5250bd4/pygmt/helpers/decorators.py#L128-L129

In GMT CLI, only a few modules (e.g., plot, plot3d, text) support varying transparency.

**Review by seisman — APPROVED** [review]
Looks good to me.

File: pygmt/src/plot.py [hunk 1]
```diff
@@ -204,7 +204,7 @@ def plot(self, x=None, y=None, data=None, size=None, direction=None, **kwargs):
     {p}
     {t}
         *transparency* can also be a 1d array to set varying transparency
-        for symbols.
+        for symbols, but it is only valid if using x/y.
```
Comment (weiji14) [inline pygmt/src/plot.py:207]:
```suggestion
        for symbols, but this option is only valid if using x/y.
```

File: pygmt/src/plot3d.py [hunk 2]
```diff
@@ -168,7 +168,7 @@ def plot3d(
     {p}
     {t}
         *transparency* can also be a 1d array to set varying transparency
-        for symbols.
+        for symbols, but it is only valid if using x/y.
```
Comment (weiji14) [inline pygmt/src/plot3d.py:171]:
```suggestion
        for symbols, but this option is only valid if using x/y.
```

**Review by weiji14 — COMMENTED** [review]
Just one tiny suggestion :)

> In GMT CLI, only a few modules (e.g., plot, plot3d, text) support varying transparency.

Seems like we're missing the varying transparency feature for `text`, maybe open an issue for this, and we can get someone to work on it (after #1121).

**Review by weiji14 — APPROVED** [review]

**Comment by core-man** [comment]
> > In GMT CLI, only a few modules (e.g., plot, plot3d, text) support varying transparency.
> 
> Seems like we're missing the varying transparency feature for `text`, maybe open an issue for this, and we can get someone to work on it (after #1121).

We have varying transparency feature for `text`, no?
https://github.com/GenericMappingTools/pygmt/blob/5125224d8e0a0fd40544f854cabed59e1833c8bf/pygmt/src/text.py#L200-L204

https://github.com/GenericMappingTools/pygmt/blob/5125224d8e0a0fd40544f854cabed59e1833c8bf/pygmt/tests/test_text.py#L324-L327

Or you mean improving the documentation for the `transparency` option in `text?
https://github.com/GenericMappingTools/pygmt/blob/5125224d8e0a0fd40544f854cabed59e1833c8bf/pygmt/src/text.py#L149

File: pygmt/src/plot3d.py [hunk 3]
```diff
@@ -168,7 +168,7 @@ def plot3d(
     {p}
     {t}
         *transparency* can also be a 1d array to set varying transparency
-        for symbols.
+        for symbols, but this option is only valid if using x/y.
```
Comment (weiji14) [inline pygmt/src/plot3d.py:171]: Actually for plot3d, it should be `x/y/z`.

```suggestion
        for symbols, but this option is only valid if using x/y/z.
```

File: pygmt/src/text.py [hunk 4]
```diff
@@ -147,6 +147,8 @@ def text_(
     {f}
     {p}
     {t}
+        *transparency* can also be a 1d array to set varying transparency
+        for texts, but this option is only valid if using x/y.
```
Comment (weiji14) [inline pygmt/src/text.py:151]: Similarly, should be `x/y/text` here.
```suggestion
        for texts, but this option is only valid if using x/y/text.
```

**Comment by core-man** [comment]
> Oh yes, forgot that varying transparency for text was added in #716. Could you please add it to the documentation for `text` then? And change the title of this PR too after.

Done in `505ec41` and `c3d0223`. But it seems that the `transparency` parameter in `text` should be polished a little, e.g., raise a **GMTInvalidInput** if 1d array `transparency` is not used with x/y.

**Comment by weiji14** [comment]
> > Oh yes, forgot that varying transparency for text was added in #716. Could you please add it to the documentation for `text` then? And change the title of this PR too after.
> 
> Done in `505ec41` and `c3d0223`. But it seems that the `transparency` parameter in `text` should be polished a little, e.g., raise a **GMTInvalidInput** if 1d array `transparency` is not used with x/y.

Yes, but that can be done in a separate Pull Request, let's keep this one focused on just the docstring (and unit test). I think @seisman is doing a bit of work on refactoring the `text` function at #1121 so best not to cause conflicts.

P.S. The ubuntu-latest - Python 3.9 / NumPy 1.20 tests are failing due to an unrelated reason at #1278. I'll see if I can come up with a quick fix, if not, it should be fine to merge (as an admin) since the other tests are passing.

**Pull request merged**